src/Tes.SDK/TesClient.cs

```diff
@@ -0,0 +1,180 @@
+            return result!.Id;
+        }
+
+        /// <inheritdoc/>
+        public async Task<TesTask> GetTaskAsync(string taskId, TesView view, CancellationToken cancellationToken)
+        {
+            var response = await SendRequestAsync(GetRequest(HttpMethod.Get, $"/v1/tasks/{taskId}", query: $"view={view}"), cancellationToken: cancellationToken);
+            return await DeserializeAsync<TesTask>(response.Content, cancellationToken)!;
+        }
+
+        /// <inheritdoc/>
+        public async Task CancelTaskAsync(string taskId, CancellationToken cancellationToken)
+        {
+            await SendRequestAsync(GetRequest(HttpMethod.Post, $"/v1/tasks/{taskId}:cancel"), cancellationToken: cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public async IAsyncEnumerable<TesTask> ListTasksAsync(TesView view, [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            string? pageToken = null;
+
+            do
+            {
+                var query = $"view={view}{(string.IsNullOrWhiteSpace(pageToken) ? string.Empty : $"&pageToken={pageToken}")}";
+                var response = await SendRequestAsync(GetRequest(HttpMethod.Get, "/v1/tasks", query), cancellationToken: cancellationToken);
+                var tesListTasksResponse = await DeserializeAsync<TesListTasksResponse>(response.Content, cancellationToken);
+
+                if (tesListTasksResponse?.Tasks?.Count > 0)
+                {
+                    foreach (var task in tesListTasksResponse.Tasks)
+                    {
+                        yield return task;
+                    }
+
+                    pageToken = tesListTasksResponse.NextPageToken;
+                }
+                else
+                {
+                    yield break;
+                }
+            } while (pageToken != null && !cancellationToken.IsCancellationRequested);
+        }
+
+        /// <inheritdoc/>
+        public async Task<TesTask> CreateAndWaitTilDoneAsync(TesTask tesTask, CancellationToken cancellationToken)
+        {
+            var taskId = await CreateTaskAsync(tesTask, cancellationToken);
+            var retryPolicy = Policy
+                .Handle<Exception>()
+                .WaitAndRetryAsync(60, _ => TimeSpan.FromSeconds(15));
+
+            while (true)
+            {
+                var task = await retryPolicy.ExecuteAsync(() => GetTaskAsync(taskId, TesView.MINIMAL, cancellationToken));
+
+                if (!task.IsActiveState())
+                {
+                    return await GetTaskAsync(taskId, TesView.FULL, cancellationToken);
+                }
+
+                await Task.Delay(TimeSpan.FromSeconds(1), cancellationToken);
+            }
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    _httpClient.Dispose();
```

Comment:
Quick digression here :)  Does this make sense?  The caller provided the HttpClient, so it doesn't seem like this class' responsibility to dispose of it.  The Microsoft docs say that `HttpClient` is intended to be instantiated once and reused throughout the life of an application.  Now - having said that, I slightly disagree about only one HttpClient, because then one has to be careful that the entire app doesn't modify the default headers.  In this case, I originally use a private static HttpClient, but then I couldn't figure out how to mock it without using putting in the constructor.  Thoughts?

Reply:
I now see the constructor where its passed in, but I don't believe were using it anywhere (except inside of this class) so it's not currently accessible to the caller to dispose.

I'll address that rn, though